packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -242,19 +243,39 @@ export class SummarizerNode implements IRootSummarizerNode {
         );
         return { latestSummaryUpdated: true, wasSummaryTracked: false, snapshot: snapshotTree };
     }
-
+/**
+ * Called when we get an ack from the server for a summary we've just sent. Updates the reference state of this node
+ * from the state in the pending summary queue.
+ * @param proposalHandle - Handle for the current proposal.
+ * @param referenceSequenceNumber -  reference sequence number of sent summary.
+ */
     protected refreshLatestSummaryFromPending(
         proposalHandle: string,
         referenceSequenceNumber: number,
     ): void {
         const summaryNode = this.pendingSummaries.get(proposalHandle);
         if (summaryNode === undefined) {
-            // This should only happen if parent skipped recursion AND no prior summary existed.
-            assert(
-                this._latestSummary === undefined,
-                0x1a6 /* "Not found pending summary, but this node has previously completed a summary" */,
-            );
-            return;
+            if (this._latestSummary !== undefined) {
+                this.wipSummaryLogger?.sendTelemetryEvent({
+                    eventName: "SummaryNodeWithoutPendingSummary",
+                    proposalHandle,
+                    pendingSummary: this._latestSummary.waitingForPendingSummary,
+                    referenceSequenceNumber,
+                });
+
+                // A data store is probably loading other data stores asynchronously or in response
+                // to some event/trigger which will cause a few data stores to be loaded in between summaries.
+                // We will simply re-use the summary node that was just created when the DataStore
+                // was realized (parentLatestSummary.createForChild), and we can safely continue processing.
+                // The only check is to make sure the reference number from the lastSummary is smaller
+                // than the one returned by the ack.
+                // https://dev.azure.com/fluidframework/internal/_workitems/edit/645
+                assert(this._latestSummary.referenceSequenceNumber < referenceSequenceNumber,
```

Comment:
Now that I think of it. When we summarize we go from container runtime to all the individual datastores to all the individual DDS's. When we are looking at the children, we are only looking at the individual DDSs